R/uvhydrograph-data.R

```diff
@@ -25,6 +25,21 @@ parseCorrectionsByMonth <- function(reportObject, seriesName, month) {
   return(corrections)
 }
 
+#' Parse Secondary Corrections
+#' @description depending on the report configuration, corrections might come from a reference or upchain series
+#' @param reportObject the report to render
+#' @return corrections list from the correct series
+parseSecondaryCorrectionsByMonth <- function(reportObject, month) {
+  hasReferenceSeries <- hasReferenceSeries(reportObject)
+  hasUpchainSeries <- hasUpchainSeries(reportObject)
+  
+  if(hasReferenceSeries) {
+    corrections <- parseCorrectionsByMonth(reportObject, "referenceSeriesCorrections", month)
+  } else {
```

Comment:
I think this should be an `else if(hasUpchainSeries)` in case it also does not have an upchain series. Or if that's handled by the parseCorrectionsByMonth function then perhaps this entire function could be reduced to something like:

```
corrections <- parseCorrectionsByMonth(reportObject, "referenceSeriesCorrections", month)

if(isEmptyOrBlank(corrections)){
    corrections <- parseCorrectionsByMonth(reportObject, "upchainSeriesCorrections", month)
}
```